src/GovUk.Education.ExploreEducationStatistics.Data.Services/SubjectMetaService.cs

```diff
@@ -335,5 +333,114 @@ private static string OrderLocationAttributes(LocationAttributeNode node)
                 : await _releaseSubjectRepository.GetReleaseSubjectForLatestPublishedVersion(subjectId) ??
                   new Either<ActionResult, ReleaseSubject>(new NotFoundResult());
         }
+
+        private async Task<Either<ActionResult, Unit>> ValidateFiltersForSubject(
+            Guid subjectId,
+            List<FilterUpdateViewModel> requestFilters)
+        {
+            var filters = await _filterRepository.GetFiltersIncludingItems(subjectId);
+            return AssertCollectionsAreSameIgnoringOrder(filters,
+                requestFilters,
+                filter => filter.Id,
+                requestFilter => requestFilter.Id,
+                "Requested filters do not match subject filters").OnSuccess(_ =>
+            {
+                var requestMap = requestFilters.ToDictionary(filter => filter.Id);
+                var results = filters.Select(filter =>
+                        ValidateFilterGroupsForSubject(filter, requestMap[filter.Id].FilterGroups))
+                    .ToList();
+
+                var failure = results.FirstOrDefault(result => result.IsLeft)?.Left;
```

Comment:
I think we have a helper method for this somewhere... will pop it on here if I find it

Reply:
I've added an `OnSuccessAll` and a `OnSuccessVoid` to `EitherExtensions` so that I can get all the comparisons and then return `OnSuccessAll().OnSuccessVoid()`. These are non-async extension methods as opposed to our other task extensions.

Reply:
:+1: